### ISSUE TITLE
L043: Applying sqlfluff fix to certain CASE WHEN statements generates invalid SQL

### ISSUE THREAD

**Issue by kdw2126** [sqlfluff/sqlfluff#2340]
## Expected Behaviour
When a CASE WHEN statement is included in SQL, ```sql fix``` should not replace it with invalid SQL that does not run.

Ideally, when the following SQL is included in a file CASE WHEN var IS NULL THEN NULL ELSE var END, SQLFluff should replace it with var. 

(I appreciate how insanely silly this syntax example is, but I'm donating my embarrassing code example for the cuse.)

## Observed Behaviour
SQLFluff instead replaces some CASE WHEN statements with incomplete COALESCE statements that do not compile properly.

## Steps to Reproduce
(1). Generate a SQL file of the form:
```sql
SELECT CASE WHEN var IS NULL THEN NULL ELSE var END AS var
FROM
    model
```
(2). Set BigQuery as the dialect and run sqlfluff fix --force on this file.
It will be replaced by this:
```sql
SELECT COALESCE( AS var
FROM
    model
```

## Dialect
BigQuery

## Version
SQLFluff version is 0.9.1.
Python version is 3.8.10.
(I'm using DBT but the same issue occurs when not using dbt as a template).

## Configuration
```
[sqlfluff]
dialect = bigquery
exclude_rules = L003,L008,L011,L014,L016,L029,L031,L034

[sqlfluff:rules]
max_line_length = 120
comma_style = leading

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
capitalisation_policy = upper
```

**Comment by jpy-git** [comment]
This is L043 rather than L034.

I've got a fix for this that I'm going to split into two parts:
1. Thanks to @barrywhart's recent work improving the CASE statement grammar we can greatly simplify the fix logic in this rule which fixes this issue in the sense that it will correctly form the coalesce statement (as well as generally simplifying the rule logic):
```sql
SELECT COALESCE(var, NULL) AS var
FROM
    model
```
2. Although this satisfies the old condition for the coalesce fix this really is a slightly different case in which we should just simplify that to:
```sql
SELECT var AS var
FROM model
```
(n.b. we're not touching that alias as that it could be different from the column name in an actual statement)

Will simplify/correct the fix logic first then we can improve the rule afterwards 👍
